src/main/java/com/commercetools/sync/commons/utils/SyncUtils.java

```diff
@@ -52,6 +56,33 @@
         return reference;
     }
 
+    /**
+     * Given a reference to a resource of type {@code T}, this method
+     * checks if the reference is expanded. If it is, then it executes the {@code keyInReferenceSupplier} and returns
+     * it's result. Otherwise, it returns the supplied reference as is. Since, the reference could be {@code null}, this
+     * method could also return null if the reference was not expanded.
+     *
+     * <p>This method expects the passed supplier to either </p>
+     *
+     * @param reference              the reference of the resource to check if it's expanded.
+     * @param <T>                    the type of the resource.
+     * @param keyInReferenceSupplier the supplier to execute and return its result if the {@code reference} was
+     *                               expanded.
+     *
+     * @return returns the result of the {@code keyInReferenceSupplier} if the {@code reference} was expanded.
+     *         Otherwise, it returns the supplied reference as is.
+     */
+    @Nullable
+    public static <T> ResourceIdentifier<T> getResourceIdentifierWithKeyReplaced(
+        @Nullable final Reference<T> reference,
+        @Nonnull final Supplier<ResourceIdentifier<T>> keyInReferenceSupplier) {
```

Comment:
maybe it's better if we use ResourceIdentifier in java doc instead of the reference to make it different with getReferenceWithKeyReplaced's Javadoc. both javadocs looks same except the method name.

Reply:
but actually the javadoc mentions the reference that is passed not the returned `ResourceIdentifier`, so I am not entirely sure which part of the javadoc should be changed

Reply:
oh you are right 👍